lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -480,40 +480,44 @@ function validatePolicyResource(bucketName, policy) {
 function checkIp(value) {
     const errString = 'Invalid IP address in Conditions';
 
-    // these preliminary checks are validating the provided
-    // ip address against ipaddr.js, the library we use when
-    // evaluating IP condition keys. It ensures compatibility,
-    // but additional checks are required to enforce the right
-    // notation (e.g., xxx.xxx.xxx.xxx/xx for IPv4). Otherwise,
-    // we would accept different ip formats, which is not
-    // standard in an AWS use case.
-    try {
+    const values = Array.isArray(value) ? value : [value];
+
+    for (let i = 0; i < values.length; i++) {
+        // these preliminary checks are validating the provided
+        // ip address against ipaddr.js, the library we use when
+        // evaluating IP condition keys. It ensures compatibility,
+        // but additional checks are required to enforce the right
+        // notation (e.g., xxx.xxx.xxx.xxx/xx for IPv4). Otherwise,
+        // we would accept different ip formats, which is not
+        // standard in an AWS use case.
         try {
-            parseCIDR(value);
+            try {
+                parseCIDR(values[i]);
+            } catch (err) {
+                isValid(values[i]);
+            }
         } catch (err) {
-            isValid(value);
+            return errString; // Return immediately if an invalid IP is found
         }
-    } catch (err) {
-        return errString;
-    }
 
-    // credit to Theodore John.S
-    // Medium article: Validating IPv4 and IPv6 Addresses with Ease
-    // — Unveiling the Power of Validation in JavaScript
-    const validateIpRegex = (ip) => {
-        if (constants.ipv4Regex.test(ip)) {
-            return ip.split('.').every(part => parseInt(part, 10) <= 255);
-        }
-        if (constants.ipv6Regex.test(ip)) {
-            return ip.split(':').every(part => part.length <= 4);
-        }
-        return false;
-    };
+        // Apply the existing IP validation logic to each element
+        const validateIpRegex = (ip) => {
+            if (constants.ipv4Regex.test(ip)) {
+                return ip.split('.').every(part => parseInt(part, 10) <= 255);
+            }
+            if (constants.ipv6Regex.test(ip)) {
+                return ip.split(':').every(part => part.length <= 4);
+            }
+            return false;
+        };
 
-    if (validateIpRegex(value) === true) {
-        return null;
+        if (validateIpRegex(values[i]) !== true) {
+            return errString; // Return immediately if an invalid IP is found
```

Comment:
I would say this specific comment is not mandatory as it says what the code does